core/src/main/scala/org/apache/spark/streaming/eventhubs/EventHubRDD.scala

```diff
@@ -72,16 +69,21 @@ class EventHubRDD(
         throw new Exception(s"$eventHubNameAndPartition cannot return data, the trace is" +
           s" ${receivingTrace.toList}")
       }
-      val receivedEventsItr = eventHubClient.receive(expectedEventNumber - receivedBuffer.size)
-      if (receivedEventsItr == null) {
-        // no more messages
-        return receivedBuffer.toList
-      }
-      val receivedEvents = receivedEventsItr.toList
+      val receivedEvents = eventHubClient.receive(expectedEventNumber - receivedBuffer.size).
+        toList
       receivingTrace += receivedEvents.length
       cnt += 1
       receivedBuffer ++= receivedEvents
+
+      receivingTrace += receivedEvents.length
+      if(receivedEvents.length > 0 && eventHubClient.partitionRuntimeInformation != null) {
+        if (receivedEvents.last.getSystemProperties.getSequenceNumber ==
+          eventHubClient.partitionRuntimeInformation.getLastSequenceNumber) {
```

Comment:
you need to keep `receivedEvents == null` to ensure that we have something old to prevent regression (since we have no control on EventHubs server behavior